### PR TITLE
Disable Quick Edit Mode by default on Windows Consoles, Add runFlag checking to ZMQ and Console threads

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
* Disables Quick Edit Mode (Mark) in Windows Consoles to prevent accidental clicks in the window from freezing the server

* Uses runFlag to signal exit to ZMQ and Console threads to properly exit without SIGABRT
## Steps to test these changes
Windows consoles (cmd, VS debug window) should not register mouse highlight or clicks unless the user right-clicks to "Mark"

### CODE REVIEW

**Comment by zach2good** [comment]
Could `int runFlag` be changed to `std::atomic<bool> runFlag` or something, not necessarily for safety, but to signal what it is and how it's used

**Comment by claywar** [comment]
Done, should be good to go

File: src/common/console_service.h [hunk 1]
```diff
@@ -19,6 +19,9 @@
 ===========================================================================
 */
 
+#ifndef _CONSOLE_SERVICE_H_
+#define _CONSOLE_SERVICE_H_
```
Comment (zach2good) [inline src/common/console_service.h:23]: No need for header guards like this, #pragma once does the same thing, it's just not part of the standard (or something)

File: src/common/kernel.cpp [hunk 2]
```diff
@@ -46,7 +46,8 @@
 #endif
 #endif
 
-int    runflag = 1;
+std::atomic<bool> runFlag = true;
```
Comment (zach2good) [inline src/common/kernel.cpp]: Maybe rename this gRunFlag? Since it's mega-global

**Review by zach2good — APPROVED** [review]

File: src/common/console_service.h (hunk 1 above)
Comment (claywar) [reply src/common/console_service.h:23]: Was failing on GCC10 without these

File: src/common/kernel.cpp (hunk 2 above)
Comment (claywar) [reply src/common/kernel.cpp]: Will do 👍

**Pull request merged**